src/dataDisplay/banner.jsx

```diff
@@ -1,43 +1,198 @@
-'use strict';
-
-import React, { Component } from 'react';
-import BannerItem from './bannerItem';
-import ClassNames from 'classnames';
-import Header from '../atoms/header';
-import Icon from '../dataDisplay/icon';
+import {
+    isUndefined,
+    isFunction,
+    forEachRight,
+} from 'lodash';
 import { Portal } from 'react-portal';
+import ClassNames from 'classnames';
 import PropTypes from 'prop-types';
+import React from 'react';
 import ReactDOM from 'react-dom';
+import BannerItem from './bannerItem';
+import Header from '../atoms/header';
+import Icon from './icon';
+import domUtils from '../utils/domUtils';
+import withTheme from '../styles/withTheme';
+
+const propTypes = {
+    children: PropTypes.node,
+    className: PropTypes.string,
+    id: PropTypes.oneOfType([
+        PropTypes.number,
+        PropTypes.string,
+    ]).isRequired,
+    isOpen: PropTypes.bool.isRequired,
+    level: PropTypes.oneOf([
+        'error',
+        'purple',
+        'secondary',
+        'success',
+        'teal',
+        'warning',
+    ]),
+    levelIcon: PropTypes.string,
+    message: PropTypes.string,
+    onAfterClose: PropTypes.func,
+    onClose: PropTypes.func.isRequired,
+    onOpen: PropTypes.func,
+    style: PropTypes.shape({}),
+    theme: PropTypes.shape({
+        zIndex: PropTypes.shape({
+            banner: PropTypes.number,
+        }),
+    }).isRequired,
+    title: PropTypes.string,
+    topPosition: PropTypes.number,
+    type: PropTypes.oneOf([
+        'alert',
+        'notification',
+    ]),
+};
+
+const defaultProps = {
+    children: undefined,
+    className: undefined,
+    level: undefined,
+    levelIcon: undefined,
+    message: undefined,
+    onAfterClose: undefined,
+    onOpen: undefined,
+    style: undefined,
+    title: undefined,
+    topPosition: undefined,
+    type: undefined,
+};
 
-class Banner extends Component {
+class Banner extends React.Component {
```

Comment:
Should it be a `PureComponent`?  Or too much regression risk in making that change?

Reply:
@groberts314  If we want to use the `PureComponent` - no any on-fly objects/functions should be used in `render()` method. But we have such style object, so, not sure that we have to rewrite the `render()` at the moment.

Reply:
Roger that.  Thanks for setting me straight! :-)